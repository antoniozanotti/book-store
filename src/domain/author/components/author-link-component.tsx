import Link from "next/link";
import { AuthorLinkComponentInterface } from "./author-link-component-interface";

export function AuthorLinkComponent({
  author,
  children,
  ...props
}: AuthorLinkComponentInterface) {
  const link = `/author/${author.page.slug}`;
  
  return (
    <Link href={link} title={author.name} rel="author" className={props.className}>
      {children}
    </Link>
  );
}
