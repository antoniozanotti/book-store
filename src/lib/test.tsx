import React, { ReactElement, ReactNode } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return <div data-testid="container">{children}</div>;
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";
import { usePathname } from "next/navigation";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
    notFound: () => null,
    usePathname: () => null,
    useSearchParams: () => null,
  };
});
