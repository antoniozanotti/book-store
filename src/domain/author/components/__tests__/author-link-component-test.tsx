import { render } from "@/lib/test";
import { AuthorLinkComponent } from "../author-link-component";
import { authorsWithPageMock } from "../../__tests__/authors-with-page-mock";

describe("AuthorLinkComponent", () => {
  it("should render", async () => {
    const { container } = render(
      <AuthorLinkComponent author={authorsWithPageMock[0]}>
        Teste
      </AuthorLinkComponent>
    );
    expect(container).toMatchSnapshot();
  });
});
