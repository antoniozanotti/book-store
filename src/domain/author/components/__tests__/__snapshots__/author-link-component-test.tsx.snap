// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorLinkComponent should render 1`] = `
<div>
  <div
    data-testid="container"
  >
    <a
      href="/author/kevin-kwan"
      rel="author"
      title="Kevin Kwan"
    >
      Teste
    </a>
  </div>
</div>
`;
